jena-arq/src/main/java/org/apache/jena/riot/RDFParserRegistry.java

```diff
@@ -206,14 +206,24 @@ public ReaderRIOT create(Lang language, ParserProfile profile) {
     private static class ReaderRIOTFactoryThrift implements ReaderRIOTFactory {
         @Override
         public ReaderRIOT create(Lang language, ParserProfile profile) {
-            return new ReaderRDFThrift() ;
+            return new ReaderRDFThrift(profile) ;
         }
     }
 
     private static class ReaderRDFThrift implements ReaderRIOT {
+        private final ParserProfile profile; 
+        public ReaderRDFThrift(ParserProfile profile) { this.profile = profile; }
+
         @Override
         public void read(InputStream in, String baseURI, ContentType ct, StreamRDF output, Context context) {
-            BinRDF.inputStreamToStream(in, output) ;
+            try {
+                BinRDF.inputStreamToStream(in, output);
+            } catch (RiotException ex) {
+                if ( profile != null && profile.getErrorHandler() != null )
+                    profile.getErrorHandler().error(ex.getMessage(), -1, -1);
+                else
+                    System.err.println(ex.getMessage());
```

Comment:
Should it go to a `Logger` instead? Or is it expected to be written to stderr?

Reply:
Thanks for seeing that. `Logger` in some way would be better

RDFThrift is a bit different because the parser does not signal errors or warnings like the other text-syntax parsers. Such raw binary is supposed to be "right" and errors are actually low level encoding problems that Thrift will catch and complain about. But each case must throw an exception regardless of the errorhandler.